File: Day6/C/Ques23.c
#include <stdio.h>
int main()
{
    int dec, bin[32], i = 0, count = 0;
    printf("Enter a Decimal no:");
    scanf("%d", &dec);
    int a = dec;
    while (dec > 0)
    {
        bin[i] = dec % 2;
        dec = dec / 2;
        i++;
    }
    printf("%d into Binary: ", a);
    for (int j = i - 1; j >= 0; j--)
    {
        printf("%d", bin[j]);
        if (bin[j] == 1)
        {
            count++;
        }
    }
    printf("\nThe count of set bits in a number is: %d", count);
    return 0;
}

File: Day6/Python/Ques23.py
n=(int(input("Enter a decimal number:")))
bin=[]
a=c=0
while n>0:
    rem=n%2
    bin.append(rem)
    n=n//2
    a +=1
for i in bin:
    if i==1:
        c+=1
print("The set bit of the number is:",c)